build.sh

```diff
@@ -29,7 +29,7 @@ declare fname_openssl="openssl*.tar.gz"
 declare fname_libevent="libevent*.zip"
 declare fname_jsoncpp="jsoncpp*.zip"
 declare fname_boost="boost*.tar.gz"
-declare fname_openssl_down="openssl-1.1.1d.tar.gz"
+declare fname_openssl_down="openssl-1.1.1i.tar.gz"
```

Comment:
Do you have sufficient test with 1.1i?

Reply:
Actually no. I don't have any idea to make sure 1.1.1i make everything right but it passed build.sh ExecutionTesting process and I use the .so/.a with [python client](https://github.com/apache/rocketmq-client-cpp) in some simple scenarios (push/pull consumer and producer) on arm/x86_64. It all works fine. Maybe some more tests should be done but I can't find any documents or resource.

Reply:
Thank you for your honesty, this pr will be followed up by the community for a long time until a convincible security test. I'm happy to follow up on it. If you have compilation problems, you could use the patch to bypass software defects :-)

Reply:
Thanks. That's the best way to ensure stability.

Reply:
There are many existing [cves in 1.1.1d](https://nvd.nist.gov/vuln/search/results?form_type=Advanced&results_type=overview&search_type=all&cpe_vendor=cpe%3A%2F%3Aopenssl&cpe_product=cpe%3A%2F%3A%3Aopenssl&cpe_version=cpe%3A%2F%3Aopenssl%3Aopenssl%3A1.1.1d)

Reply:
OpenSSL is supposed to be dynamically linked. SDK should not care about the version of OpenSSL that is eventually used as long as they share compatible ABI.